From 2674b1e9435e47229dc43c2cdca92ec38252835b Mon Sep 17 00:00:00 2001
From: "Ooi, Joyce" <joyce.ooi@intel.com>
Date: Sat, 30 Nov 2019 02:42:55 +0800
Subject: [PATCH 080/120] net: ethernet: altera: add Intel FPGA QSE Driver
 Makefile and Kconfig

commit e6f27e60a996d7bbce4d3c69bb389c69d3612db0 from
https://github.com/altera-opensource/linux-socfpga.git

This patch adds Makefile and Kconfig for Intel FPGA Quad Speed Ethernet
(QSE) driver

Signed-off-by: Ooi, Joyce <joyce.ooi@intel.com>
Integrated-by: Abhishek Paliwal <paliwal.abhishek@windriver.com>
---
 drivers/net/ethernet/Makefile        |  2 +-
 drivers/net/ethernet/altera/Kconfig  | 28 ++++++++++++++++++++++++++--
 drivers/net/ethernet/altera/Makefile | 17 +++++++++++++----
 3 files changed, 40 insertions(+), 7 deletions(-)

diff --git a/drivers/net/ethernet/Makefile b/drivers/net/ethernet/Makefile
index 7b5bf9682066..c830b0a7a9c4 100644
--- a/drivers/net/ethernet/Makefile
+++ b/drivers/net/ethernet/Makefile
@@ -11,7 +11,7 @@ obj-$(CONFIG_NET_VENDOR_AGERE) += agere/
 obj-$(CONFIG_NET_VENDOR_ALACRITECH) += alacritech/
 obj-$(CONFIG_NET_VENDOR_ALLWINNER) += allwinner/
 obj-$(CONFIG_NET_VENDOR_ALTEON) += alteon/
-obj-$(CONFIG_ALTERA_TSE) += altera/
+obj-$(CONFIG_NET_ALTERA_ETH) += altera/
 obj-$(CONFIG_NET_VENDOR_AMAZON) += amazon/
 obj-$(CONFIG_NET_VENDOR_AMD) += amd/
 obj-$(CONFIG_NET_XGENE) += apm/
diff --git a/drivers/net/ethernet/altera/Kconfig b/drivers/net/ethernet/altera/Kconfig
index 6dec7094cb4b..432b656bc900 100644
--- a/drivers/net/ethernet/altera/Kconfig
+++ b/drivers/net/ethernet/altera/Kconfig
@@ -1,11 +1,35 @@
 # SPDX-License-Identifier: GPL-2.0-only
+config NET_ALTERA_ETH
+	tristate "Altera Ethernet drivers"
+	depends on HAS_DMA
+	---help---
+	  This driver supports ethernet drivers built around Altera
+	  Ethernet and DMA IP.
+
+	  If you have a network (Ethernet) IP belonging to this class, say Y.
+
+if NET_ALTERA_ETH
+
 config ALTERA_TSE
 	tristate "Altera Triple-Speed Ethernet MAC support"
-	depends on HAS_DMA
+	select PTP_1588_CLOCK
 	select PHYLIB
-	imply PTP_1588_CLOCK
 	---help---
 	  This driver supports the Altera Triple-Speed (TSE) Ethernet MAC.
 
 	  To compile this driver as a module, choose M here. The module
 	  will be called alteratse.
+
+config INTEL_FPGA_QSE_LL
+	tristate "Intel FPGA Quad-Speed Low Latency Ethernet MAC support"
+	select PTP_1588_CLOCK
+	select PHYLINK
+	---help---
+	  This driver supports the Intel FPGA Low Latency Quad-Speed (QSE)
+	  Ethernet MAC.
+
+	  To compile this driver as a module, choose M here. The module
+	  will be called intel_fpga_qse_ll.
+
+endif
+
diff --git a/drivers/net/ethernet/altera/Makefile b/drivers/net/ethernet/altera/Makefile
index 4834e972e906..21a9de506e4d 100644
--- a/drivers/net/ethernet/altera/Makefile
+++ b/drivers/net/ethernet/altera/Makefile
@@ -3,7 +3,16 @@
 # Makefile for the Altera device drivers.
 #
 
-obj-$(CONFIG_ALTERA_TSE) += altera_tse.o
-altera_tse-objs := altera_tse_main.o altera_tse_ethtool.o \
-		   altera_msgdma.o altera_sgdma.o altera_utils.o \
-		   intel_fpga_tod.o altera_msgdma_prefetcher.o
+obj-$(CONFIG_NET_ALTERA_ETH) += altera_eth.o
+altera_eth-objs := altera_utils.o intel_fpga_tod.o altera_eth_dma.o \
+		   altera_sgdma.o altera_msgdma.o \
+		   altera_msgdma_prefetcher.o
+ifeq ($(CONFIG_ALTERA_TSE),y)
+	obj-$(CONFIG_ALTERA_TSE) += altera_tse.o
+	altera_tse-objs := altera_tse_main.o altera_tse_ethtool.o
+endif
+ifeq ($(CONFIG_INTEL_FPGA_QSE_LL),y)
+	obj-$(CONFIG_INTEL_FPGA_QSE_LL) += intel_fpga_qse_ll.o
+	intel_fpga_qse_ll-objs := intel_fpga_qse_ll_main.o intel_fpga_qse_ll_ethtool.o
+endif
+
-- 
2.31.1

